### PR TITLE
EVAKA-4240: employee creates reservations

### PR DESCRIPTION
**Pull request by Joosakur**
#### Summary

Included
- Employee can create reservations (one child at a time for now)
- Citizen can also overwrite absences made by employees (as long as they are after the deadline/locking date)
- If weekly repetition contains empty days, existing reservations on those days are removed
- Citizen-frontend validation utils moved under lib-common (many of the changed files are just import changes)

Not included
- This PR does not yet prevent making reservation for a day excluded in weekly service times (citizen nor employee)

### CODE REVIEW

**Review by jaahon — APPROVED** [review]

File: service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt [hunk 1]
```diff
@@ -71,6 +74,86 @@ class AttendanceReservationController(private val ac: AccessControl) {
...
     }
+
+    @PostMapping
+    fun postReservations(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @RequestBody body: List<DailyReservationRequest>
+    ) {
+        Audit.AttendanceReservationEmployeeCreate.log(targetId = body.map { it.childId }.toSet().joinToString())
+        body.map { it.childId }.toSet().forEach { childId ->
```
[the remark this answers is not in the document]
Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt:85]: it's meant to remove duplicates so we don't query permissions for same child many times

File: frontend/src/citizen-frontend/calendar/DayView.tsx [hunk 2]
```diff
@@ -260,8 +257,10 @@ function useEditState(
         .map(({ child, startTime, endTime }) => ({
           childId: child.id,
           date: date,
-          startTime,
-          endTime
+          reservation: {
+            startTime,
+            endTime
+          }
```
[the remark this answers is not in the document]
Comment (Joosakur) [reply frontend/src/citizen-frontend/calendar/DayView.tsx:263]: ah, yes

File: frontend/src/employee-frontend/components/unit/unit-reservations/ReservationModalSingleChild.tsx [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    setFormData((prev) => ({
+      ...prev,
+      ...updated
+    }))
+  }
+
+  const [errors, setErrors] = useState<ReservationErrors>({
+    startDate: undefined,
+    endDate: undefined
+  })
```
[the remark this answers is not in the document]
Comment (Joosakur) [reply frontend/src/employee-frontend/components/unit/unit-reservations/ReservationModalSingleChild.tsx:78]: do you mean `const errors = useMemo(..., [formData])` or something else

[the remark this answers is not in the document]
Comment (jaahon) [reply frontend/src/employee-frontend/components/unit/unit-reservations/ReservationModalSingleChild.tsx:78]: Sure, if you think it needs to be memoized.

File: service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaahon) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt:85]: Right, I was thinking the payload was different. Sorry.

[the remark this answers is not in the document]
Comment (jaahon) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt:85]: But you could do the conversion to a set only once to save some energy 😄

**Pull request merged**